### PR TITLE
feat(message): allow customizing content_type without breaking body decoding (introduce transport_content_type header)

### LINKED ISSUE #34: Do not use the header content_type to build the body when consuming
For example, publishing XML in a message body uses a `Text` body, but the message will have the `text/plain` content_type.

The `content_type` need to stay customizable for the user.

### PR DESCRIPTION
**Pull request by Niktux**
Closes #34

if GTM, go to tag 3.0.0

### CODE REVIEW

File: src/Messages/Message.php [hunk 1]
```diff
@@ -86,9 +88,26 @@ public function disallowSilentDropping()
 
     public function getContentType()
     {
+        if($this->contentType !== false)
```
Comment (Niktux) [inline src/Messages/Message.php]: ! empty

Comment (lebris) [inline src/Messages/Message.php]: Fix 697952687920afce77e40d5f5ed2c93d232f44f3

**Comment by lebris** [comment]
As we discussed, this way puzzle/amqp require the header `transport_content_type` to be setted in order to be consumable, otherwise the `BodyFactory` return an instance of `NullBody`.

This make puzzle/amqp not compatible with a message sent without puzzle/amqp.

It would be better to allow the user to give the mapping `content_type` => `Body`.

**Comment by Niktux** [comment]
Memo :

- MessageAdapterFactory
- include BodyFactory
- allow customization (content_type -> body implementation)

**Pull request closed without merging**